packages/run-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -78,22 +93,58 @@ export const makeVaultManager = (
     async getCollateralQuote() {
       // get a quote for one unit of the collateral
       const displayInfo = await E(collateralBrand).getDisplayInfo();
-      const decimalPlaces = (displayInfo && displayInfo.decimalPlaces) || 0n;
+      const decimalPlaces = displayInfo?.decimalPlaces || 0n;
       return E(priceAuthority).quoteGiven(
         AmountMath.make(collateralBrand, 10n ** Nat(decimalPlaces)),
         runBrand,
       );
     },
   };
 
-  // A Map from vaultKits to their most recent ratio of debt to
-  // collateralization. (This representation won't be optimized; when we need
-  // better performance, use virtual objects.)
+  let vaultCounter = 0;
+
+  // A store for of vaultKits prioritized by their collaterization ratio.
   //
-  // sortedVaultKits should only be set once, but can't be set until after the
+  // It should be set only once but it's a `let` because it can't be set until after the
   // definition of reschedulePriceCheck, which refers to sortedVaultKits
-  let sortedVaultKits;
+  // XXX mutability and flow control, could be refactored with a listener
```

Comment:
I chose XXX to say "here's a known issue" but I don't think we necessarily have to schedule addressing it. I've updated the comment to clarify:
```
// XXX misleading mutability and confusing flow control; could be refactored with a listener
```

Reply:
Yeah, didn't implement it yet. I've got some more refactoring tasks coming and it might fit into one of those.

Gist is that right now we're delaying construction of A until the callback B is available that need A in scope. What we can do instead is wire up the listening after they're both created.
```
const A = makeA();
const B = makeB();
A.onEventBCaresAbout(B);
```